crates/red_knot/src/module.rs

```diff
@@ -237,36 +241,31 @@ where
 /// Resolves the module id for the given path.
 ///
 /// Returns `None` if the path is not a module in `sys.path`.
-// WARNING!: It's important that this method takes `&mut self`. Without, the implementation is prone to race conditions.
-// Note: This won't work with salsa because `Path` is not an ingredient.
-pub fn path_to_module<Db>(db: &mut Db, path: &Path) -> Option<Module>
```

Comment:
I think that's actually fine. There's no risk for deadlocks because the function only ever holds on to a single lock (and not multiple ones). Any mutations happen through `resolve_module` which must be idempotent. The only overhead here is that we might end up calling `resolve_module` unnecessary but that's fine. 

we could even consider not making this method a query. Searching the root paths is cheap. I think the only place where we safe some performance is that we avoid the `canoncialize` call. So that's why I think it's still worth keeping the cache for now.